starlite/utils/serialization.py

```diff
@@ -20,4 +23,87 @@ def default_serializer(value: Any) -> Any:
         return value.get_secret_value()
     if isinstance(value, (PurePath, PurePosixPath)):
         return str(value)
-    raise TypeError("unsupported type")
+    if isinstance(value, UUID):
+        return str(value)
+    if isinstance(value, (datetime, date, time)):
+        return value.isoformat()
+    raise TypeError(f"Unsupported type: {type(value)!r}")
+
+
+def dec_hook(type_: Any, value: Any) -> Any:
+    """Transform values non-natively supported by `msgspec`
+
+    Args:
+        type_: Encountered type
+        value: Value to coerce
+
+    Returns:
+        A `msgspec`-supported type
+    """
+    if issubclass(type_, BaseModel):
+        return type_(**value)
+    raise TypeError(f"Unsupported type: {type(value)!r}")
+
+
+_msgspec_json_encoder = msgspec.json.Encoder(enc_hook=default_serializer)
+_msgspec_json_decoder = msgspec.json.Decoder(dec_hook=dec_hook)
+_msgspec_msgpack_encoder = msgspec.msgpack.Encoder(enc_hook=default_serializer)
+_msgspec_msgpack_decoder = msgspec.msgpack.Decoder(dec_hook=dec_hook)
+
+
+def encode_json(obj: Any, enc_hook: Optional[Callable[[Any], Any]] = default_serializer) -> bytes:
+    """Encode a value into JSON.
+
+    Args:
+        obj: Value to encode
+        enc_hook: Optional callable to support non-natively supported types
+
+    Returns:
+        JSON as bytes
+    """
+    if enc_hook is None or enc_hook is default_serializer:
+        return _msgspec_json_encoder.encode(obj)
+    return msgspec.json.encode(obj, enc_hook=enc_hook)
+
+
+def decode_json(raw: Union[str, bytes]) -> Any:
+    """Decode a JSON string/bytes into an object.
+
+    Args:
+        raw: Value to decode
+
+    Returns:
+        An object
+    """
+    if isinstance(raw, str):
+        raw = raw.encode("utf-8")
```

Comment:
Are strings ever passed to `decode_json` in a hot path? Or is this just for compatibility? We could support this natively in `msgspec` (which would be slightly faster than doing this), but decoding from a strings should be avoided for performance reasons when possible.